crates/extensions/tedge_log_manager/src/actor.rs

```diff
@@ -0,0 +1,269 @@
+use super::LogManagerConfig;
+
+fan_in_message_type!(LogInput[MqttMessage, FsWatchEvent] : Debug);
+fan_in_message_type!(LogOutput[MqttMessage]: Debug);
+
+pub struct LogManagerActor {
+    config: LogManagerConfig,
+    pub plugin_config: LogPluginConfig,
+    mqtt_publisher: LoggingSender<MqttMessage>,
+    messages: SimpleMessageBox<LogInput, NoMessage>,
+    http_proxy: ClientMessageBox<HttpRequest, HttpResult>,
+}
+
+#[async_trait]
+impl Actor for LogManagerActor {
+    fn name(&self) -> &str {
+        "LogManager"
+    }
+
+    async fn run(&mut self) -> Result<(), RuntimeError> {
+        self.reload_supported_log_types().await?;
+
+        while let Some(event) = self.messages.recv().await {
+            match event {
+                LogInput::MqttMessage(message) => {
+                    self.process_mqtt_message(message).await?;
+                }
+                LogInput::FsWatchEvent(event) => {
+                    self.process_file_watch_events(event).await?;
+                }
+            }
+        }
+        Ok(())
+    }
+}
+
+impl LogManagerActor {
+    pub fn new(
+        config: LogManagerConfig,
+        plugin_config: LogPluginConfig,
+        mqtt_publisher: LoggingSender<MqttMessage>,
+        messages: SimpleMessageBox<LogInput, NoMessage>,
+        http_proxy: ClientMessageBox<HttpRequest, HttpResult>,
+    ) -> Self {
+        Self {
+            config,
+            plugin_config,
+            mqtt_publisher,
+            messages,
+            http_proxy,
+        }
+    }
+
+    pub async fn process_mqtt_message(
+        &mut self,
+        message: MqttMessage,
+    ) -> Result<(), LogManagementError> {
+        if self.config.logfile_request_topic.accept(&message) {
+            match LogRequestPayload::try_from(message.clone()) {
+                Ok(request) => {
+                    if (request.status.eq(&CommandStatus::Executing)
+                        || request.status.eq(&CommandStatus::Init))
+                        && !self.config.current_operations.contains(&message.topic.name)
+                    {
+                        info!("Log request received: {request:?}");
+                        self.config
+                            .current_operations
+                            .insert(message.topic.name.clone());
+                        self.handle_logfile_request_operation(&message.topic, &request)
+                            .await?;
+                    }
+                }
+                Err(err) => {
+                    error!("Incorrect log request payload: {}", err);
+                }
+            }
+        } else {
+            error!(
+                "Received unexpected message on topic: {}",
+                message.topic.name
+            );
+        }
+
+        Ok(())
+    }
+
+    pub async fn handle_logfile_request_operation(
+        &mut self,
+        topic: &Topic,
+        request: &LogRequestPayload,
+    ) -> Result<(), LogManagementError> {
```

Comment:
This fuction returns always `Ok(())`, so doesn't need to return `Result`.

Reply:
> This fuction returns always `Ok(())`, so doesn't need to return `Result`.

No this method might fail with a `ChannelError` if the operation status cannot be passed over to the MQTT actor.

So, this method must return a `Result` but with a more specific error type.

```suggestion
    ) -> Result<(), ChannelError> {
```